modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/DefaultPolicyInstance.java

```diff
@@ -0,0 +1,157 @@
+   * @param template policy template from which the instance will be created. Non null
+   * @param parametrization parameters used to configure the created instance. Non null
+   * @param serviceRepository repository of available services. Non null.
+   * @param classLoaderRepository contains the registered classloaders that can be used to load serialized classes. Non null.
+   */
+  public DefaultPolicyInstance(Application application, PolicyTemplate template,
+                               PolicyParametrization parametrization, ServiceRepository serviceRepository,
+                               ClassLoaderRepository classLoaderRepository) {
+    this.application = application;
+    this.template = template;
+    this.parametrization = parametrization;
+    this.serviceRepository = serviceRepository;
+    this.classLoaderRepository = classLoaderRepository;
+  }
+
+  private void initPolicyContext() {
+    if (policyContext == null) {
+      ArtifactContextBuilder artifactBuilder =
+          newBuilder().setArtifactType(APP)
+              .setArtifactName(parametrization.getId())
+              .setConfigurationFiles(getResourcePaths(template.getDescriptor().getConfigResourceFiles()))
+              .setExecutionClassloader(template.getArtifactClassLoader().getClassLoader())
+              .setServiceRepository(serviceRepository)
+              .setClassLoaderRepository(classLoaderRepository);
+
+      artifactBuilder.withServiceConfigurator(customizationService -> customizationService
+          .overrideDefaultServiceImpl(MuleProperties.OBJECT_POLICY_MANAGER_STATE_HANDLER,
+                                      application.getMuleContext().getRegistry()
+                                          .lookupObject(MuleProperties.OBJECT_POLICY_MANAGER_STATE_HANDLER)));
+
+      try {
+        policyContext = artifactBuilder.build();
+        policyContext.getMuleContext().start();
+      } catch (MuleException e) {
+        throw new IllegalStateException("Cannot create artifact context for the policy instance", e);
+      }
+    }
+  }
+
+  private String[] getResourcePaths(File[] configResourceFiles) {
+    List<String> paths = new ArrayList<>();
+    for (File configResourceFile : configResourceFiles) {
+      paths.add(configResourceFile.getAbsolutePath());
+    }
+
+    return paths.toArray(new String[0]);
+  }
+
+  @Override
+  public List<Policy> findSourceParameterizedPolicies(
+                                                      PolicyPointcutParameters policyPointcutParameters) {
+
+    initPolicyInstance();
+
+    return policyInstance.getSourcePolicyChain()
+        .map(operationChain -> asList(new Policy(operationChain, parametrization.getId())))
+        .orElse(emptyList());
+  }
+
+  private void initPolicyInstance() {
+
+    if (policyInstance == null) {
+      synchronized (this) {
+        initPolicyContext();
+
+        try {
+          policyInstance = policyContext.getMuleContext().getRegistry().lookupObject(
```

Comment:
definitively weird to have this in this class. Sounds like a third party should be injected with this default policy instance and then wrap it in this other class